File: test/extensions/language/expressions/cpp/DivisionOperatorTest.cpp
#include "expressions/division_operator/DivisionFunction.h"
#include "gtest/gtest.h"

namespace expressions
{
namespace division_operator
{

TEST(DivisionOperatorTest, divideFloatByInt)
{
    const DivisionFunction fun(10, 2);
    ASSERT_NEAR(fun.funcDivideFloatByInt(), 3.33333, 1e-5);
}

TEST(DivisionOperatorTest, divideIntByFloat)
{
    const DivisionFunction fun(10, 2);
    ASSERT_NEAR(fun.funcDivideIntByFloat(), 3.33333, 1e-5);
}

TEST(DivisionOperatorTest, divideFloatByFloat)
{
    const DivisionFunction fun(10, 2);
    ASSERT_NEAR(fun.funcDivideFloatByFloat(), 3.33333, 1e-5);
}

TEST(DivisionOperatorTest, divideIntByInt)
{
    const DivisionFunction fun(10, 2);
    ASSERT_EQ(fun.funcDivideIntByInt(), 3);
}

TEST(DivisionOperatorTest, negativeRounding)
{
    const DivisionFunction fun;
    ASSERT_EQ(fun.funcPositiveByNegativeRounding(), -1);
    ASSERT_EQ(fun.funcNegativeByPositiveRounding(), -1);
    ASSERT_EQ(fun.funcNegativeByNegativeRounding(), 1);
}

} // namespace division_operator
} // namespace expressions

File: test/extensions/language/expressions/python/DivisionOperatorTest.py
import Expressions


class DivisionOperatorTest(Expressions.TestCase):
    def testDivideFloatByInt(self):
        fun = self.api.DivisionFunction(10, 2)
        self.assertAlmostEqual(fun.divide_float_by_int(), 3.33333, places=5)

    def testDivideIntByFloat(self):
        fun = self.api.DivisionFunction(10, 2)
        self.assertAlmostEqual(fun.divide_int_by_float(), 3.33333, places=5)

    def testDivideFloatByFloat(self):
        fun = self.api.DivisionFunction(10, 2)
        self.assertAlmostEqual(fun.divide_float_by_float(), 3.33333, places=5)

    def testDivideIntByInt(self):
        fun = self.api.DivisionFunction(10, 2)
        self.assertAlmostEqual(fun.divide_int_by_int(), 3, places=5)
